### PR TITLE
Fix EventEmitter.once issue and add EventEmitter removeAllListeners patch.

### PR DESCRIPTION
**Pull request by JiaLiPassion**
I found several issues of patching EventEmitter's listener related methods.

with zone.js 0.6.22 release, addListener, prependListener, removeListener are patched in this commit.
https://github.com/angular/zone.js/commit/100b82bbac4d720be0cb7ead11398e7fe95d1132

1. issue #484 EventEmitter.once was not removed properly.
the issue can be described as the following test case.

` emitter.once('test', shouldNotRun);
   emitter.removeListener('test', shouldNotRun);`

in 0.6.22, the once listener was not removed.

the reason is when removeListener call util.ts, findExistingRegisteredTask method, it will look for the correct event task by check the eventTask.data.handler === handler, but if the handler is attached by EventEmitter.once method, the handler will not attached directly, instead it will be attached through a _onceWrap (https://github.com/nodejs/node/blob/master/lib/events.js#L286), and keep the original handler to the onceWrap.listener property, so we should not only check the data.handler but also the data.handler.listener(https://github.com/angular/zone.js/pull/500/commits/2ee4711e96cad3b27bce79ea9290800c929986f1#diff-2e7dc744ee86a31cde46d41cda5e42c0R714)

2. issue #491 removeListeners was not patched
the issue can be described as the following test case.
`emitter.on('test', expectZoneA);
  emitter.on('test', expectZoneA);
  emitter.removeAllListeners('test');`

in 0.6.22, the listener was not removed.

the reason is based on the issue #491,  in the  zone.js 0.6.22 release,  to fix #430,  addListener, removeListener, prependListener, and listeners have been patched, but removeAllListeners is not, In issue #491, socket-io will call removeAllListeners to cache and remove the listeners on request event when attach server, but because the removeAllListeners is not patched and handled correctly by zone(the response listener was added to event tasks because the addListener is patched but not removed because removeAllListeners are not patched), the request event will be handled multiple times and cause express response throw "can't set header after response sent " error.
I add a patch of removeAllListeners based on the #430 release.

3. when add listener by prependListener, the listener should be added at the beginning of the listener array.
the issue can be described as the following test case.

`emitter.on('test', listenerA);
  emitter.prependListener('test', listenerB);
  expect(emitter.listeners('test')).toEqual([listenerB, listenerA]);`

in 0.6.22, the order is still listenerA, listenerB

The reason is in 0.6.22 prependListener do the same logic with addListener.

4. use the newest fetch to fix Object #<Response> has no method 'blob' error

 please review the source code of those 2 issues, Thank you.

### CODE REVIEW

**Comment by jakubmal** [comment]
Nice! I might be mistaken, but you should not commit dist versions, they are updated, when publishing a new version, right?

**Comment by JiaLiPassion** [comment]
@jakubmal  Yes, you are right, I will modify my Pull Request. Thank you.

**Comment by JiaLiPassion** [comment]
I modify the package.json to use the newest fetch to avoid travis CI build error(Object #<Response> has no method 'blob').

File: package.json [hunk 1]
```diff
@@ -64,6 +64,6 @@
     "ts-loader": "^0.6.0",
     "tslint": "^3.15.1",
     "typescript": "^2.0.2",
-    "whatwg-fetch": "^1.0.0"
+    "whatwg-fetch": "https://github.com/jimmywarting/fetch.git"
```
Comment (mhevery) [inline package.json]: Why is this change needed? This does not seem right.
